internal/formats/template/format.go

```diff
@@ -0,0 +1,46 @@
+package template
+
+import (
+	"io"
+
+	"github.com/anchore/syft/internal/formats/syftjson"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+const ID sbom.FormatID = "template"
+
+func Format() sbom.Format {
+	return OutputFormat{}
+}
+
+// implementation of sbom.Format interface
+// to make use of format options
+type OutputFormat struct {
+	templateFilePath string
+}
+
+func (f OutputFormat) ID() sbom.FormatID {
+	return ID
+}
+
+func (f OutputFormat) Decode(reader io.Reader) (*sbom.SBOM, error) {
+	return nil, sbom.ErrDecodingNotSupported
+}
+
+func (f OutputFormat) Encode(output io.Writer, s sbom.SBOM) error {
+	tmpl, err := makeTemplateExecutor(f.templateFilePath)
+	if err != nil {
+		return err
+	}
+
+	doc := syftjson.ToFormatModel(s)
+	return tmpl.Execute(output, doc)
+}
+
+func (f OutputFormat) Validate(reader io.Reader) error {
+	return sbom.ErrValidationNotSupported
+}
+
+func (f OutputFormat) WithTemplate(filePath string) sbom.Format {
+	return OutputFormat{templateFilePath: filePath}
```

Comment:
it's a little odd that this is worded like a functional option, but creates a new object like a builder pattern.

I think this fits a little better as a setter for a field on the instance it was called on, something like
```
f.SetTemplatePath(path)
``` 
(side steps the need to create a new instance of the format).

Reply:
I see now. I've changed the func signature to that.